src/CMakeLists.txt

```diff
@@ -427,10 +436,46 @@ add_custom_command(TARGET ${EXECUTABLE}
 # to make the .lst file consistent across different build paths.
 #
 # If you come up with a better solution, please remove this :D
+#		
+	COMMAND bash -c "${CMAKE_OBJDUMP} --prefix=${SRC_DIR} --prefix-strip=$$(echo $${PWD//\\//} | tr -cd '/' | wc -c) -S -d -l -w ${EXECUTABLE} | tr -s ' ' | cut -d' ' -f2- | sed 's=${SRC_DIR}/./==' > ${EXECUTABLE}.lst"
+	BYPRODUCTS ${EXECUTABLE}.lst
+)
+else()
+# Generate lst file
+add_custom_command(TARGET ${EXECUTABLE}
```

Comment:
Is `else()` and below the same as above? Could we put the ` CMAKE_HOST_WIN32` to only encompass the delta in the _horrible looking command_

Reply:
you can't nest the if/else into the `add_custom_command` ( i tried it and it didn't work 😅), but I suppose I could do a 

```
if(win)
 horrible_command = windows horrible command
else()
 horrible_command = normal horrible command
endif()

add_custom_command(
.
.
.
horrible_command
)
```

Reply:
Good suggestion! I changed it and it works on both windows and mac!